sstables/mp_row_consumer.hh

```diff
@@ -974,13 +978,9 @@ public:
 
     // See the RowConsumer concept
     void push_ready_fragments() {
-        if (!_mf_filter || _mf_filter->out_of_range()) {
-            _reader->on_out_of_clustering_range();
-            return;
-        }
-
```

Comment:
The code fragment that gained the _mf_filter-> thing already knows _mf_filter is engaged.

Reply:
@denesb `push_ready_fragments()` only requires `_m_filter` to be set when `_stored_tombstone` is engaged. And in this case, we really need _m_filter. We cannot decide what to do with `_stored_tombstone` without `_m_filter`. It's a logic error if `_m_filter` is not engaged. Hence the assert. The previous behavior of exiting when `!_mf_filter` would yield incorrect read results in case `_stored_tombstone` is engaged. So I don't think we're less safe after the patch.

Reply:
@denesb Do you accept this explanation? If so, please resolve.

Reply:
I accept it but I don't know how to resolve my comment. I don't see such an option.

Reply:
@denesb there's a button here ---v

Reply:
Not on my box:
![image](https://user-images.githubusercontent.com/1389273/60086820-7a7b9300-9744-11e9-8ddf-3fb75394d78d.png)

Reply:
Maybe because I didn't start a review, just added a comment.

Reply:
Strange, maybe github expects the commit author to resolve the conversation (and not the one that raised the question), and I only see the button due to unrelated superpowers.